src/configure/helper/devOps/azureDevOpsHelper.ts

```diff
@@ -115,6 +117,18 @@ export class AzureDevOpsHelper {
         };
     }
 
+    private async getAgentQueueId(organizationName: string, projectName: string, poolName: string): Promise<number> {
+        let queues = await this.azureDevOpsClient.getAgentQueues(organizationName, projectName);
+
+        for(let queue of queues) {
+            if(queue.pool && queue.pool.name && queue.pool.name.toLowerCase() === poolName.toLowerCase()) {
+                return queue.id;
+            }
+        }
+
+        throw new Error(util.format(Messages.noAgentQueueFound, poolName));
```

Comment:
Should we return first entry instead of failing here. Anyways the queue would get overridden with the value specified in yaml, so does not make sense to fail here

Reply:
Agree. Will do.